GQI - Add info about paged ad hoc data sources (#3023)
SkylineCommunications/dataminer-docs: user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/API_Reference/GQIPage.md

| Property | Type | Required | Description |
| -------- | ---- | -------- | ----------- |
| Rows | [GQIRow](xref:GQI_GQIRow)[] | Yes | Rows of the page. |
| HasNextPage | boolean  | No | `true` if there are more pages that GQI can fetch, otherwise 'false' (default). |

> [!TIP]

> > Learn how `HasNextPage` can be used to apply paged data retrieval through following example: [paged data retrieval example](xref:GQI_IGQIDataSource#paged-data-retrieval)

Reviewer: Might be good to mention here briefly why you would care.
The link to the example can be integrated in the text.

```suggestion
> When dealing with large amounts of data, it is recommended to spread the data across multiple pages. See [this example](xref:GQI_IGQIDataSource#paged-data-retrieval) on how the `HasNextPage` property can be used to enable paged data retrieval.
```

| -------- | ---- | -------- | ----------- |
| Rows | [GQIRow](xref:GQI_GQIRow)[] | Yes | Rows of the page. |
| HasNextPage | boolean  | No | `True` if GQI can fetch additional pages, `False` if the current page is the last page. |

> > [!TIP]

Reviewer: Feel like this tip has a lot of words but not really to the point.

Suggestion (to be chatGPTified):
Often not all data is required immediately to produce a query result. Therefore, consider splitting up rows in separate pages so they can be fetched and processed individually as needed to improve speed and memory efficiency. For example: [Link to example]

> To enhance performance and efficiency, consider restricting the data returned by using the `HasNextPage` boolean of a `GQIPage`. Often, not all data from an ad hoc data source is essential for the visualization. Utilizing paged data retrieval makes sure the implementation only retrieves the necessary pages until GQI accumulates sufficient data. This approach offers several benefits:
> - Speed: Fetching all data immediately in your implementation can be inefficient. By retrieving only the required pages, you optimize processing speed.
> - Memory Efficiency: Limiting the data retrieved helps conserve memory resources. Unnecessary data won't burden the system, allowing it to operate more efficiently.

### Examples

#### Paged data retrieval

> The following example illustrates how the `HasNextPage` boolean can be manipulated to indicate that there are additional pages. If `HasNextPage` is set to `False`, GQI might call `GetNextPage` once more, until it has enough data.

Reviewer: ```suggestion
The following example illustrates how the `HasNextPage` boolean can be used to indicate that there are additional pages. If `HasNextPage` is set to `true`, GQI may call `GetNextPage` again when more data is needed.
```

```csharp

> private int pageCounter = 0;

Reviewer: Should be incremented at some point

private int pageSize = 100;

public GQIPage GetNextPage(GetNextPageInputArgs args)
{
    var rows = FetchData(pageCounter, pageSize);

>     var hasNextPage = rows.Length == pageSize;

Reviewer: Maybe this needs a comment to explain that we assume there is more data if we get a full page.

    return new GQIPage(rows)
    {
        HasNextPage = hasNextPage;
    };
}

> private GQIRow[] FetchData(int page, int pageSize){

Reviewer: Can rename this method a bit, so it's immediately obvious in the code above what it does.
Also, can be simplified by not providing a page size since:

1. at this point in time, there is never a reason that it would change
2. the page number has no meaning if the page size isn't fixed 

```suggestion
private GQIRow[] FetchPagedData(int pageNumber){
```

>     // Retrieve the data starting from 'page'

Reviewer: ```suggestion
    // Fetch page 'pageNumber' containing at most 'pageSize' rows
```
